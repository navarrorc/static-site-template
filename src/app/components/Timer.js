import React, { Component } from "react";
import styled from "react-emotion";
import colors from "./colors";

const Container = styled("div")`
  color: ${colors.light};
  display: inline-block;
  padding: 0.25rem;
  font-size: 2rem;
  border: 5px solid ${colors.primary};

  span {
    background-color: ${colors.dark};
  }
`;

class Timer extends Component {
  // constructor(props) {
  //   super(props);
  //   this.state = { seconds: 0 };
  // } /* no need as we are using the transform-class-properties bable puglin, see the package.json file

  state = {
    seconds: 0
  };

  tick() {
    this.setState(state => ({
      seconds: state.seconds + 1
    }));
  }

  componentDidMount() {
    this.interval = setInterval(() => this.tick(), 1000);
  }

  componentWillUnmount() {
    clearInterval(this.interval);
  }

  render() {
    return (
      <Container style={{ margin: "1rem" }}>
        <span>Seconds: {this.state.seconds}</span>
      </Container>
    );
  }
}

export default Timer;
