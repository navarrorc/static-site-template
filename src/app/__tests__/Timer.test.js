import React from "react";
import { create } from "react-test-renderer";
import Timer from "../components/Timer";

test("snapshot", () => {
  const c = create(<Timer />);
  expect(c.toJSON()).toMatchSnapshot();
});
