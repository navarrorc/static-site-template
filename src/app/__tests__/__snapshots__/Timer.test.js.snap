// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  style={
    Object {
      "margin": "1rem",
    }
  }
>
  Seconds: 
  0
</div>
`;
